third_party/imx-secure-enclave/BUILD.gn

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+declare_args() {
+  NOT_SUPPORTED = 0
+  ROM = 1
+  FMW = 2
+
+  MAJOR_VER = 1
+  MINOR_VER = 0
+
+  PLAT_PATH = "src/plat/ele"
+  PLAT_COMMON_PATH = "src/common"
+}
+
+config("libele_config") {
```

Comment:
use a buildconfig_header instead of a config. Passing in a lot of command line arguments via configs is not ideal.

That would also make you not pass in things like `DEBUG` which probably overrides others.